src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationFactory.cs

```diff
@@ -161,6 +164,17 @@ private void SetContentRoot(IWebHostBuilder builder)
             }
         }
 
+        private bool TrySetContentRootUsingEnvironmentVariables(IWebHostBuilder builder)
+        {
+            var expectedVariableName = "AspNetCore.Mvc.WebRoot-" + typeof(TEntryPoint).Assembly.GetName().Name;
+            var variableValue = Environment.GetEnvironmentVariable(expectedVariableName);
```

Comment:
Environment variables are already a built in part of the config system. Read from config rather than directly from env.

Reply:
@Tratcher How would that work?

Reply:
`IWebHostBuilder.GetSetting(expectedVariableName);`, minus the ASPNETCORE_ prefix.